add more examples checkvalidity (#4580)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst

     - Example

   * - Repeated point

>      - This error happens when a given node is repeated.

Reviewer: ```suggestion
     - This error happens when a given vertex is repeated.
```

Reviewer: We should reserve the use of "node" to the 0D elements of network topology.
As long as we don't have a better word, we will have to stick to "vertex".

          :align: center

   * - Duplicate rings
     - This error happens when a part of MultiPolygon geometry is inside a hole of a MultiPolygon geometry. 

>      - .. figure:: img/geos_dupl_rings.png

Reviewer: ```suggestion

     - .. figure:: img/geos_dupl_rings.png
```

Reviewer: Not sure if this is needed, but just in case....

          :align: center 

   * - Too few points in geometry component
     - 
     -

   * - Invalid coordinate

>      - This error happens when A point don't have a single coordinate pair. The coordinate pair don't contains a latitude value and a longitude value in that order. 

Reviewer: ```suggestion
     - For a point geometry, this error happens when the geometry does not
       have a proper coordinate pair.
       The coordinate pair does not contain a latitude value and a longitude
       value in that order. 
```

          :align: center

   * - Duplicate rings

>      - This error happens when a part of MultiPolygon geometry is inside a hole of a MultiPolygon geometry. 

Reviewer: "Duplicate rings" does not match this explanation.  Is it the error message that should be changed, or this explanation.
"Duplicate rings" should translate to: "This error happens when two rings (interior or exterior) are identical"

Reviewer: Sorry, did not notice this in the first place.

Author: Sorry I miss this.

>      - This error happens when a part of MultiPolygon geometry is inside a hole of a MultiPolygon geometry. 

Reviewer: ```suggestion
     - This error happens when two rings (exterior or interior) of a polygon geometry
       are identical
```

Reviewer: @jonnyforestGIS , this suggestion was in a comment. I made a proper "suggestion" out of it. When it has been committed, I think this is ready to merge.

     -

   * - Line %1 contains %n duplicate node(s) at %2 

>      - This error happens when a given node is repeated.

Reviewer: ```suggestion
     - This error happens when consecutive points ona line have the same
       coordinates.
```

Reviewer: Sorry, missing blank...

     - .. figure:: img/geos_rep_point.png
          :align: center 

   * - Segments %1 and %2 of line %3 intersect at %4 

>      - This error happens when segments lines intersects between them.

Reviewer: ```suggestion
     - This error happens when a line self intersects (two segments of the
       line intersect each other).
```

     - .. figure:: img/qgis_seg_line_int.png
          :align: center

   * - Ring self-intersection 

>      - This error happens when a geometry touches itself and generates a ring.

Reviewer: ```suggestion
     - This error happens when an inner ring / boundary (island) of a polygon
       geometry intersects itself.
```

     -

   * - Line %1 contains %n duplicate node(s) at %2 
     - This error happens when a given node is repeated.
     - .. figure:: img/geos_rep_point.png
          :align: center 

   * - Segments %1 and %2 of line %3 intersect at %4 
     - This error happens when a line self intersects (two segments of the
       line intersect each other).
     - .. figure:: img/qgis_seg_line_int.png
          :align: center

   * - Ring self-intersection 

>      - This error happens when an inner ring / boundary (island) of a polygon

Reviewer: ```suggestion
     - This error happens when an outer or inner (island) ring / boundary of a polygon
```

Reviewer: My first suggestion was not correct.  I assume that this applies to both outer and inner rings.
